pkg/cloudprovider/providers/vsphere/vsphere_util.go

```diff
@@ -251,6 +253,94 @@ func getPbmCompatibleDatastore(ctx context.Context, dc *vclib.Datacenter, storag
 	return datastore, err
 }
 
+func getDatastoresForZone(ctx context.Context, dc *vclib.Datacenter, nodeManager *NodeManager, selectedZones []string) ([]*vclib.DatastoreInfo, error) {
```

Comment:
I do not expect the zone to datastores mapping to change at all once configured correctly. If we compute and cache zones for every datastore in Virtual Center in a map, this method can just compute the datastores in selectedZones very quickly. Don't you agree?

Reply:
Agree. And since this being a performance improvement, hope its fine to be taken as a separate work item under this issue - https://github.com/kubernetes/kubernetes/issues/73955